glide-core/src/client/mod.rs

```diff
@@ -525,11 +525,31 @@ impl Client {
                     Some(ResponsePolicy::AllSucceeded),
                 ));
                 let mut cmd = redis::cmd("AUTH");
+                if let Ok(Some(username)) = self.get_username().await {
+                    cmd.arg(username);
+                }
                 cmd.arg(password);
                 self.send_command(&cmd, Some(routing)).await
             }
         }
     }
+
+    /// Returns the username if one was configured during client creation. Otherwise, returns None.
+    async fn get_username(&mut self) -> RedisResult<Option<String>> {
+        match &mut self.internal_client {
+            ClientWrapper::Cluster { client } => match client.get_username().await {
+                Ok(Value::SimpleString(username)) => Ok(Some(username)),
+                Ok(Value::Nil) => Ok(None),
+                Ok(other) => unreachable!("Expected SimpleString or Nil, got: {:?}", other),
```

Comment:
don't use unreachable as it would panic and crash the whole client. Instead, return:
RedisError::from((
                ErrorKind::ClientError,
                "Expected SimpleString or Nil, got: {other:?}"
            ))